### PR TITLE
Refactor storage subsystem

### PR DESCRIPTION
**Pull request by rojkov**
* Don't maintain `maxTabId` in `DBManager`. Initialize tab models' `nextTabId` once upon their creation instead.
* Encapsulate `Link` creation into `DBWorker` (`TabModel` doesn't have to know about this implementation detail). Also this makes possible  dropping `TabModel`'s `nextLinkId` members and their maintenance code.
* `TabModel`'s `nextTabId` property is used nowhere - drop it from public API.
* Reduce data exchange between SQLite and DBWorker by using SQL inner joins.
* Don't include `Link` to `Tab` as a member to avoid maintaining linkId in `Tab`.
* Fully cover the storage subsystem with unit tests.

### CODE REVIEW

**Comment by tworaz** [comment]
LGTM

File: src/history/persistenttabmodel.cpp [hunk 1]
```diff
@@ -57,10 +57,15 @@ void PersistentTabModel::tabsAvailable(QList<Tab> tabs)
         emit countChanged();
     }
 
-    int maxTabId = DBManager::instance()->getMaxTabId();
+    int maxTabId(0);
+    foreach (Tab tab, tabs) {
```
Comment (rainemak) [inline src/history/persistenttabmodel.cpp]: ```foreach (const Tab &tab, tabs) {```

**Comment by rojkov** [comment]
Good catch! Also I've maid the ::tabsAvailable() slot less hungry for copying tabs.

File: tests/auto/tst_dbmanager/tst_dbmanager.cpp [hunk 2]
```diff
@@ -11,125 +11,482 @@
```
Comment (rainemak) [inline tests/auto/tst_dbmanager/tst_dbmanager.cpp]: Cleanup tests.xml a bit. Remove ```-platform wayland-egl -iterations 10``` part from the tst_dbmanager case.

**Comment by rainemak** [comment]
Added one comment about the tests.xml. Other than that LGTM now.

**Comment by rojkov** [comment]
Updated tests.xml

**Comment by rojkov** [comment]
Amended commit messages.

**Pull request merged**